pandas/tests/groupby/test_sample.py

```diff
@@ -116,14 +116,18 @@ def test_groupby_sample_without_n_or_frac():
     tm.assert_series_equal(result, expected)
 
 
-def test_groupby_sample_with_weights():
+@pytest.mark.parametrize(
+    "index, expect_index",
+    [(["w", "x", "y", "z"], ["w", "w", "y", "y"]), ([3, 4, 5, 6], [3, 3, 5, 5])],
+)
+def test_groupby_sample_with_weights(index, expect_index):
```

Comment:
Can you add issue number in a comment as the first line of this function (i.e. `GH ##39927`)